File: math/fatorial/fatorial.py
# encoding: utf-8

"""
Cálculo de fatorial
Autor:
    ?
Colaborador:
    Bruno Lara Tavares <bruno.exz@gmail.com>
Tipo:
    math
Descrição:
    Calcula o fatorial de um número
Complexidade:  
    ?
Dificuldade:
    facil
"""


def fatorial(b):
	return 1 if b <= 1 else b*fatorial(b-1)

print fatorial(6)

File: math/fatorial/fatorial.rb
# -*- encoding: utf-8 -*-

"""
Cálculo de fatorial
Autor:
    ?
Tipo:
    math
Descrição:
    Calcula o fatorial de um número
Complexidade:  
    Pior caso: ?
    Melhor caso: ?
Dificuldade:
    facil
"""


def fatorial(b)
	b <= 1 ? 1 : b*fatorial(b-1)
end

puts fatorial(6)
